src/GraphQL/Execution/ExecutionStrategy.cs

```diff
@@ -134,13 +177,136 @@ private void SetSubFieldNodes(ExecutionContext context, ObjectExecutionNode pare
             }
 
             parent.SubFields = subFields;
+
+            fields.Clear();
+            System.Threading.Interlocked.CompareExchange(ref context.ReusableFields, fields, null);
+        }
+
+        /// <summary>
+        /// Returns a <see cref="FieldType"/> for the specified AST <see cref="Field"/> within a specified parent
+        /// output graph type within a given schema. For meta-fields, returns the proper meta-field field type.
+        /// </summary>
+        protected virtual FieldType GetFieldDefinition(ISchema schema, IObjectGraphType parentType, Field field)
+        {
+            if (field.Name == schema.SchemaMetaFieldType.Name && schema.Query == parentType)
+            {
+                return schema.SchemaMetaFieldType;
+            }
+            if (field.Name == schema.TypeMetaFieldType.Name && schema.Query == parentType)
+            {
+                return schema.TypeMetaFieldType;
+            }
+            if (field.Name == schema.TypeNameMetaFieldType.Name)
+            {
+                return schema.TypeNameMetaFieldType;
+            }
+
+            if (parentType == null)
+            {
+                throw new ArgumentNullException(nameof(parentType), $"Schema is not configured correctly to fetch field '{field.Name}'. Are you missing a root type?");
+            }
+
+            return parentType.GetField(field.Name);
+        }
+
+        /// <inheritdoc/>
+        public virtual Fields GetSubFields(ExecutionContext context, ExecutionNode node)
+        {
+            return node.Field?.SelectionSet?.Selections?.Count > 0
+                ? CollectFieldsFrom(context, node.FieldDefinition.ResolvedType, node.Field.SelectionSet)
+                : null;
+        }
+
+        /// <summary>
+        /// Before execution, the selection set is converted to a grouped field set by calling CollectFields().
+        /// Each entry in the grouped field set is a list of fields that share a response key (the alias if defined,
+        /// otherwise the field name). This ensures all fields with the same response key included via referenced
+        /// fragments are executed at the same time.
+        /// <br/><br/>
+        /// <see href="http://spec.graphql.org/June2018/#sec-Field-Collection"/> and <see href="http://spec.graphql.org/June2018/#CollectFields()"/>
+        /// </summary>
+        /// <param name="context">The execution context.</param>
+        /// <param name="specificType">The graph type to compare the selection set against.</param>
+        /// <param name="selectionSet">The selection set from the document.</param>
+        /// <param name="fields">A list to append the collected list of fields to; if null, a new list will be created.</param>
```

Comment:
```suggestion
        /// <param name="fields">A list to append the collected list of fields to; if <see langword="null"/>, a new list will be created.</param>
```
> A list to append

Not a list, dictionary.

Reply:
```suggestion
        /// <param name="fields">A dictionary to append the collected list of fields to; if <see langword="null"/>, a new dictionary will be created.</param>
```

Reply:
Please confirm updated xml comment.

Reply:
OK